resource-managers/yarn/src/test/scala/org/apache/spark/network/shuffle/ShuffleTestAccessor.scala

```diff
@@ -220,4 +219,21 @@ object ShuffleTestAccessor {
       db: DB): ConcurrentMap[ExternalShuffleBlockResolver.AppExecId, ExecutorShuffleInfo] = {
     ExternalShuffleBlockResolver.reloadRegisteredExecutors(db)
   }
+
+  private def initDB(dbBackend: DBBackend, file: File): DB = {
+    if (file != null) {
+      // TODO: SPARK-38888, add rocksdb implementation.
+      dbBackend match {
+        case DBBackend.LEVELDB =>
+          val options = new org.iq80.leveldb.Options()
+          options.createIfMissing(true)
+          val factory = new org.fusesource.leveldbjni.JniDBFactory()
+          new LevelDB(factory.open(file, options))
+        case _ =>
+          throw new IllegalArgumentException("Unsupported DBBackend: " + dbBackend)
+      }
+    } else {
+      null
+    }
```

Comment:
We will end up duplicating logic between `DBProvider` and `ShuffleTestAccessor` - let us consolidate it in `DBProvider`.
This also impacts the integration test anyway

Reply:
OK, let me revert this change first and then see if there is a better way

Reply:
do some check, I think we can directly use another `initDB` and delete this one.